src/main/groovy/com/blackducksoftware/integration/hub/detect/bomtool/yarn/YarnBomTool.groovy

```diff
@@ -55,41 +68,77 @@ class YarnBomTool extends BomTool implements NestedBomTool<BomToolSearchResult>
     private BomToolSearchResult searchResult;
 
     @Override
-    public BomToolType getBomToolType() {
+    BomToolType getBomToolType() {
         BomToolType.YARN
     }
 
     @Override
-    public boolean isBomToolApplicable() {
-        BomToolSearchResult searchResult = bomToolSearcher.getBomToolSearchResult(sourcePath);
+    boolean isBomToolApplicable() {
+        BomToolSearchResult searchResult = bomToolSearcher.getBomToolSearchResult(sourcePath)
         if (searchResult.isApplicable()) {
-            this.searchResult = searchResult;
-            return true;
+            this.searchResult = searchResult
+            return true
         }
 
-        return false;
+        return false
     }
 
-    public List<DetectCodeLocation> extractDetectCodeLocations(BomToolSearchResult searchResult) {
-        final File yarnLockFile = detectFileManager.findFile(searchResult.searchedDirectory, 'yarn.lock')
-        final List<String> yarnLockText = Files.readAllLines(yarnLockFile.toPath(), StandardCharsets.UTF_8)
-        final DependencyGraph dependencyGraph = yarnPackager.parse(yarnLockText)
-        final ExternalId externalId = externalIdFactory.createPathExternalId(Forge.NPM, searchResult.searchedDirectory.canonicalPath)
-        final def detectCodeLocation = new DetectCodeLocation.Builder(getBomToolType(), searchResult.searchedDirectory.canonicalPath, externalId, dependencyGraph).build()
+    List<DetectCodeLocation> extractDetectCodeLocations(BomToolSearchResult searchResult) {
+        DependencyGraph dependencyGraph
+        DetectCodeLocation detectCodeLocation
+
+        File yarnLockFile = detectFileManager.findFile(searchResult.searchedDirectory.canonicalPath, 'yarn.lock')
+        List<String> yarnLockText = Files.readAllLines(yarnLockFile.toPath(), StandardCharsets.UTF_8)
+        String yarnExePath = findExecutablePath(ExecutableType.YARN, true, detectConfiguration.getYarnPath())
+
+
+        if (detectConfiguration.yarnProductionDependenciesOnly && StringUtils.isBlank(yarnExePath)) {
+            throw new BomToolException("Could not find the Yarn executable, can not get the production only dependencies.")
```

Comment:
Not sure how we want to handle the scenario where the User wants production only dependencies but we can not find the Yarn executable. For now I am throwing an exception rather than parsing the lock